### PR TITLE
lesson-14

### PR DESCRIPTION
**Pull request by DariaPavlyuk81**
form and functionality added

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -137,6 +134,41 @@ <h2>Connect:</h2>
...
+        id="email"
+        required 
+        autocomplete="on"
+        /><br />
+        <label for="usersMessage"> Users Message:  </label>
+        <textarea id="usersMessage"
+        name="usersMessage"
+        required ></textarea>
+        
+        /><br />
```
Comment (dindustack) [inline index.html]: Take of `/>`before the `<br />` tag

File: css/index.css [hunk 2]
```diff
@@ -259,15 +238,65 @@ a:active {
...
+#messages h2{
+  font-size:1.5 rem;
+  color:#333;
+  margin-bottom:20px;
+  text-align: center;
+}
+.message-container{
+  overflow-y:auto;
+}
+#message-list{
```
Comment (dindustack) [inline css/index.css]: Update CSS styling for message-list

.message-list {
    list-style-type: none;
    padding: 0;
    margin: 0;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.message-list li span {
    display: block;
}

.message-list > .remove-button {
    background-color: #F00;
    color: #fff;
    border: 1px solid #F00;
    padding: 5px 10px;
    border-radius: 10px;
    cursor: pointer;
    font-size: 0.875;
    margin-left: 0.5rem;
}

File: js/index.js [hunk 3]
```diff
@@ -34,23 +33,85 @@ const skillsList = skillsSection.getElementsByTagName("ul")[0];
...
+
+    //display message beneath messages section
+    const messageSection = document.getElementById("messages");
+    const messageList = messageSection.querySelector("ul");
+
+
+
+//display the message section
+messageSection.hidden = false;
+const newMessage = document.createElement("li");
```
Comment (dindustack) [inline js/index.js:71]: Add a class to li for styling

`newMessage.classList.add("message-list");`

File: js/index.js [hunk 4]
```diff
@@ -34,23 +33,85 @@ const skillsList = skillsSection.getElementsByTagName("ul")[0];
...
-//}
\ No newline at end of file
+//find the leave messages form and add the callback for submit
+const messageForms = document.getElementsByName("leave_message");
+const messageForm= messageForms[0];
+
+//Attach the listener
+messageForm.addEventListener("submit", onFormSubmit);
+
+//callback for remove button
```
Comment (dindustack) [inline js/index.js]: remove button function should come under `FormSubmit`function

File: css/index.css [hunk 5]
```diff
@@ -259,15 +238,65 @@ a:active {
...
+}
+
+form label{
+  display:block;
+  margin-bottom:5 px;
+  font-size: 1 rem;
+  color:#333;
+}
+#messages{
+  margin:20px auto;
```
Comment (dindustack) [inline css/index.css]: Add `display: none` to `#messages` styling

**Review by dindustack — CHANGES_REQUESTED** [review]
Improve the aesthetics of the form input CSS styling and review the corrections I made

File: css/index.css [hunk 6]
```diff
@@ -154,89 +178,178 @@ a:hover {
...
-      gap: 0.25rem;
-    }
+
+#messages{
+  margin:20 px auto;
+  padding: 20px;
+  border-radius: 8px;
+  background-color: #f4f4f4;
+  box-shadow: 0 0 10px rgba(0,0,0,0.1);
+  display: none;
```
Comment (dindustack) [inline css/index.css]: Add the styling below to `#messages`

`border: 1px solid black`

**Review by dindustack — CHANGES_REQUESTED** [review]
The skills and connect and form sections are missing their styling.

Good work on the responsiveness.

File: js/index.js [hunk 7]
```diff
@@ -34,23 +33,113 @@ const skillsList = skillsSection.getElementsByTagName("ul")[0];
...
+
+//Add remove Button
+newMessage.appendChild(removeButton);
+//add message
+messageList.appendChild(newMessage);
+
+//reset the form
+event.target.reset();
+
+//callback for remove button
```
Comment (dindustack) [inline js/index.js]: Please, remove lines 107 - 118, there are two functionalities with the name `onRemoveButton`

**Review by dindustack — CHANGES_REQUESTED** [review]
I have left comments for correction

File: js/index.js (hunk 7 above)
Comment (DariaPavlyuk81) [reply js/index.js]: ok, thank you

**Review by dindustack — CHANGES_REQUESTED** [review]
1. For the messages section, the content should be in a horizontal line, work on the CSS styling. 

2. For the form section, work on the CSS styling of the input field.

3. Remove empty lines of code

HTML (index.html)
1. Labels for Form Fields: Each form field should have a corresponding label. While you have included labels, ensure they are correctly associated with their respective form fields using the for attribute matching the input's id. For example, <label for="usersName">Users Name:</label> is correctly set, but make sure all labels follow this pattern.
Section Element Above Footer: The instructions required adding the new 'section' element above the 'footer' element. However, the footer is being added dynamically via JavaScript, which is not mentioned in the instructions. Consider modifying the structure in the HTML directly or adjusting the JavaScript to meet the specification.

JavaScript (index.js)

1. Variable Naming and Code Clarity: Use consistent and descriptive variable names. For instance, messageForm and messageForms[0] can be confusing. Simplify this by directly selecting the form using document.querySelector('[name="leave_message"]') to improve readability.

**Pull request merged**